src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -0,0 +1,128 @@
+  instanceTypesNames: string[];
+};
+
+const EditBootableVolumesModal: FC<EditBootSourceModalProps> = ({
+  dataSource,
+  isOpen,
+  onClose,
+  preferences,
+  instanceTypesNames,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [, setIsDropdownOpen] = React.useState(false);
+
+  const preferencesMap = useMemo(() => convertResourceArrayToMap(preferences), [preferences]);
+  const preferencesNames = Object.keys(preferencesMap);
+  const initialMetadata: BootableVolumeMetadata = {
+    labels: dataSource?.metadata?.labels,
+    annotations: dataSource?.metadata?.annotations,
+  };
+  const [metadata, setMetadata] = useState<BootableVolumeMetadata>(initialMetadata);
+
+  const setBootableVolumeMetadata = (key: string, fieldKey: string) => (value: string) =>
+    setMetadata((prevState) => ({
+      ...prevState,
+      [key]: { ...prevState[key], [fieldKey]: value },
+    }));
+
+  const onSubmit = async () => {
+    initialMetadata !== metadata &&
+      (await k8sPatch({
+        model: DataSourceModel,
+        resource: dataSource,
+        data: [
+          {
+            op: 'replace',
+            path: '/metadata/labels',
+            value: metadata.labels,
+          },
+          {
+            op: 'replace',
+            path: '/metadata/annotations',
+            value: metadata.annotations,
+          },
+        ],
+      }));
+  };
+
+  return (
+    <TabModal<K8sResourceCommon>
+      obj={dataSource}
+      isOpen={isOpen}
+      onClose={onClose}
+      headerText={t('Edit')}
+      onSubmit={onSubmit}
+    >
+      <Form>
+        <FormGroup label={t('Default preference')} isRequired>
+          <FilterSelect
+            selected={metadata?.labels?.[DEFAULT_PREFERENCE_LABEL]}
+            setSelected={setBootableVolumeMetadata('labels', DEFAULT_PREFERENCE_LABEL)}
+            options={preferencesNames}
+            groupVersionKind={VirtualMachineClusterPreferenceModelGroupVersionKind}
+            optionLabelText={t('preference')}
+          />
+        </FormGroup>
+        <FormGroup label={t('Default Instancetype')}>
+          <FilterSelect
+            selected={metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]}
+            setSelected={setBootableVolumeMetadata('labels', DEFAULT_INSTANCETYPE_LABEL)}
+            options={instanceTypesNames}
+            groupVersionKind={VirtualMachineClusterInstancetypeModelGroupVersionKind}
+            optionLabelText={t('Instancetype')}
+          />
+        </FormGroup>
+        <FormGroup label={t('Operating system name')}>
+          <Select
+            isDisabled
+            onToggle={setIsDropdownOpen}
+            selections={getPreferenceReadableOS(dataSource, preferences)}
+          />
+        </FormGroup>
+        <FormGroup label={t('Description')}>
+          <TextArea
+            value={metadata?.annotations?.description}
+            onChange={setBootableVolumeMetadata('annotations', ANNOTATIONS.description)}
```

Comment:
please add ?. before description i dont know if its always promised

Reply:
It's always promised, until we delete the appropriate key from our `ANNOTATIONS` object ;)